src/pages/settings/Payments/ChooseTransferAccountPage.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import ROUTES from '../../../ROUTES';
+import HeaderWithCloseButton from '../../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+import Navigation from '../../../libs/Navigation/Navigation';
+import styles from '../../../styles/styles';
+import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
+import KeyboardAvoidingView from '../../../components/KeyboardAvoidingView/index';
+import PaymentMethodList from './PaymentMethodList';
+import ONYXKEYS from '../../../ONYXKEYS';
+import compose from '../../../libs/compose';
+import * as paymentPropTypes from './paymentPropTypes';
+import * as PaymentMethods from '../../../libs/actions/PaymentMethods';
+import CONST from '../../../CONST';
+
+const propTypes = {
+    walletTransfer: paymentPropTypes.walletTransferPropTypes,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    walletTransfer: {},
+};
+
+class ChooseTransferAccountPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.paymentMethodSelected = this.paymentMethodSelected.bind(this);
+    }
+
+    /**
+     * Navigate to the appropriate payment method type addition screen
+     * @param {String} paymentMethodType
+     */
+    navigateToAddPaymentMethod(paymentMethodType) {
+        if (paymentMethodType === CONST.PAYMENT_METHODS.PAYPAL) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_PAYPAL_ME);
+        }
+
+        if (paymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_DEBIT_CARD);
+        }
+
+        if (paymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_BANK_ACCOUNT);
+        }
+    }
+
+    /**
+     * Go back to TransferPage with the selected bank account
+     *
+     * @param {Object} nativeEvent
+     * @param {String} accountID id of the selected account.
+     */
+    paymentMethodSelected(nativeEvent, accountID) {
```

Comment:
Is the `nativeEvent` used?

Reply:
Also this should be called something like `setPaymentMethodAndNavigate()`

Reply:
Yeah, this is used to anchor the popover menu on the `PaymentsPage`.

Reply:
I don't see where it's used in this method. Why is it being passed to this method?

Reply:
PaymentsPage was expecting the first parameter to be an event object so I had to define one for keeping the same func defintion. But I think as per your prev comment, we should divide this into two methods and the problem will be solved.